quipucords/fingerprinter/formatters.py

```diff
@@ -31,3 +35,57 @@ def convert_architecture(architecture):
     if not architecture_map.get(architecture):
         return architecture
     return architecture_map.get(architecture)
+
+
+def str_or_none(value):
+    """Normalize value to be a string (or None when "empty")."""
+    if value is None:
+        return None
+    if isinstance(value, str) and value.strip() == "":
+        return None
+    return str(value).strip()
+
+
+def float_or_none(value):
+    """Convert value to float."""
+    with suppress(ValueError, TypeError):
+        return float(value)
+    return None
+
+
+def int_or_none(value):
+    """Convert value to int."""
+    try:
+        return int(value)
+    except TypeError:
+        return None
+    except ValueError:
+        ...
+
+    with suppress(ValueError):
+        return int(float_or_none(value))
+
+    return None
+
+
+def list_or_none(value):
+    """Normalize value as list or return None."""
+    if isinstance(value, list):
+        return value
+    logger.error("value '%s' is NOT a list")
+
+
+def dict_or_none(value):
+    """Normalize value as dict or return None."""
+    if isinstance(value, dict):
+        return value
+    logger.error("value '%s' is NOT a dict")
+
+
+def date_or_none(value):
```

Comment:
This one gets a little long, but if we add any, we probably should add them all. Could define custom types, I guess, but _effort_.

```suggestion
def date_or_none(value: datetime.date | datetime.datetime) -> datetime.date | datetime.datetime | None:
```

Reply:
datetime should not be a possible output, so I only included date and None